### PR TITLE
Update unlock command

### LINKED ISSUE #7667: Update unlock command
### Description
- Update unlock command to include the events and store usage to the [LIP](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0057.md#unlock).
- Update verify logic only to check params is empty
- Check current other execute logic is aligned with the LIP

### Acceptance Criteria

- Add tests for the new events
- Update verify tests

### PR DESCRIPTION
**Pull request by mosmartin**
### What was the problem?

This PR resolves #7667

### How was it solved?

- Add verify()
- Add _isUnlockable()
- Refactor execute()
- Update constants

### CODE REVIEW

File: framework/src/modules/dpos_v2/commands/unlock.ts [hunk 1]
```diff
@@ -35,6 +47,23 @@ export class UnlockCommand extends BaseCommand {
 		this._roundLength = args.roundLength;
 	}
 
+	// TODO: Check verification logic
+	// eslint-disable-next-line @typescript-eslint/require-await
+	public async verify(context: CommandVerifyContext): Promise<VerificationResult> {
```
Comment (shuse2) [inline framework/src/modules/dpos_v2/commands/unlock.ts:52]: this LGTM

File: framework/src/modules/dpos_v2/commands/unlock.ts [hunk 2]
```diff
@@ -54,30 +83,78 @@ export class UnlockCommand extends BaseCommand {
...
+				this._isUnlockable(unlockObject, pomHeights, senderAddress, height) &&
 				isCertificateGenerated({
 					unlockObject,
 					genesisHeight,
 					maxHeightCertified,
 					roundLength: this._roundLength,
 				})
 			) {
+				// delete unlockObject from voterStore(senderAddress).pendingUnlocks
+				await voterSubstore.del(context, senderAddress);
```
Comment (shuse2) [inline framework/src/modules/dpos_v2/commands/unlock.ts:95]: this deletes the while voter data, but it should just remove the unlock object

File: framework/src/modules/dpos_v2/constants.ts [hunk 3]
```diff
@@ -36,6 +36,10 @@ export const CHAIN_ID_LENGTH = 4;
 export const LOCAL_ID_LENGTH = 4;
 export const TOKEN_ID_LENGTH = CHAIN_ID_LENGTH + LOCAL_ID_LENGTH;
 export const MAX_NUMBER_BYTES_Q96 = 24;
+export const LOCKING_PERIOD_VOTES = 26_000;
+export const LOCKING_PERIOD_SELF_VOTES = 260_000;
+export const PUNISHMENT_WINDOW_SELF_VOTES = 780_000;
+export const PUNISHMENT_WINDOW_VOTES = 260_000;
```
Comment (shuse2) [inline framework/src/modules/dpos_v2/constants.ts:42]: these contacts exist above

File: framework/src/modules/dpos_v2/commands/unlock.ts [hunk 4]
```diff
@@ -54,30 +83,78 @@ export class UnlockCommand extends BaseCommand {
...
-		if (voterData.pendingUnlocks.length === ineligibleUnlocks.length) {
-			throw new Error('No eligible voter data was found for unlocking');
+		// if (voterData.pendingUnlocks.length === ineligibleUnlocks.length) {
+		// 	throw new Error('No eligible voter data was found for unlocking');
+		// }
+		// voterData.pendingUnlocks = ineligibleUnlocks;
+		// await voterSubstore.set(context, senderAddress, voterData);
+	}
+
+	public _isUnlockable(
```
Comment (shuse2) [inline framework/src/modules/dpos_v2/commands/unlock.ts:134]: is it doing something different from the original behavior?

File: framework/src/modules/dpos_v2/commands/unlock.ts [hunk 5]
```diff
@@ -54,30 +83,78 @@ export class UnlockCommand extends BaseCommand {
...
+		height: number,
+	): boolean {
+		const { delegateAddress } = unlockObject;
+		const lastPomHeight = pomHeights[pomHeights.length - 1];
+		const lockingPeriod = delegateAddress.equals(senderAddress)
+			? LOCKING_PERIOD_SELF_VOTES
+			: LOCKING_PERIOD_VOTES;
+		const punishmentWindow = delegateAddress.equals(senderAddress)
+			? PUNISHMENT_WINDOW_SELF_VOTES
+			: PUNISHMENT_WINDOW_VOTES;
```
Comment (shuse2) [inline framework/src/modules/dpos_v2/commands/unlock.ts:147]: function to compute these exist in the utils

**Review by shuse2 — CHANGES_REQUESTED** [review]

**Review by shuse2 — APPROVED** [review]

File: framework/test/unit/modules/dpos_v2/commands/unlock.spec.ts [hunk 6]
```diff
@@ -122,6 +127,56 @@ describe('UnlockCommand', () => {
...
+				})
+				.createCommandVerifyContext();
+
+			const result = await unlockCommand.verify(verifyContext);
+
+			expect(result.status).toBe(VerifyStatus.OK);
+		});
+
+		// check if transaction params are empty
+		it('should return an error if transaction params are empty', async () => {
```
Comment (has5aan) [inline framework/test/unit/modules/dpos_v2/commands/unlock.spec.ts:148]: Pretty trivial; I guess message intend to say; 'should return an error if transaction params are not empty'

**Review by has5aan — APPROVED** [review]

File: framework/test/unit/modules/dpos_v2/commands/unlock.spec.ts (hunk 6 above)
Comment (mosmartin) [reply framework/test/unit/modules/dpos_v2/commands/unlock.spec.ts:148]: Nice catch @has5aan, you are right 😀

**Pull request merged**